### ISSUE TITLE
keyboard size is not recalculating when different keyboard types are used and AvoidSoftView is not disappearing when Keyboard is hidden

### ISSUE THREAD

**Issue by IvanIhnatsiuk** [mateusz1913/react-native-avoid-softinput#43]
### Environment

Library version: 2.3.3
OS version:  Nexus S API 29

### Affected platforms

- [X] Android
- [ ] iOS

### Current behavior

Hi, @mateusz1913! 

When you have different types of keyboard - alphabetic and numeric, `AvoidSoftView` does not recalculate the height and is sometimes still displayed when the keyboard is hidden.

https://user-images.githubusercontent.com/86000012/158979568-0f5da29e-cdf9-4327-bd10-6ea8fa20f2cf.mov

### Expected behavior

AvoidView should recalculate when the keyboard height is changed and AvoidView should disappear when the keyboard is hidden

### Reproduction

Just past this code into FormExample.tsx

```tsx
import { useFocusEffect } from "@react-navigation/native";
import React, { useCallback } from "react";
import { Image, ScrollView, StyleSheet, TextInput, View } from "react-native";
import { AvoidSoftInput } from "react-native-avoid-softinput";
import { SafeAreaView } from "react-native-safe-area-context";

import Button from "../components/Button";
import MultilineInput from "../components/MultilineInput";
import SingleInput from "../components/SingleInput";
import { styles as commonStyles } from "../consts/styles";

const icon = require("../../assets/AppIconTransparent.png");

// eslint-disable-next-line @typescript-eslint/no-empty-function
const NOOP = () => {};

export const FormExample: React.FC = () => {
  const onFocusEffect = useCallback(() => {
    AvoidSoftInput.setAdjustNothing();
    AvoidSoftInput.setEnabled(true);
    return () => {
      AvoidSoftInput.setEnabled(false);
      AvoidSoftInput.setDefaultAppSoftInputMode();
    };
  }, []);

  useFocusEffect(onFocusEffect);

  return (
    <SafeAreaView
      edges={["bottom", "left", "right"]}
      style={commonStyles.screenContainer}
    >
      <ScrollView
        bounces={false}
        contentContainerStyle={commonStyles.scrollContainer}
        contentInsetAdjustmentBehavior="always"
        overScrollMode="always"
        showsVerticalScrollIndicator={true}
        style={commonStyles.stretch}
      >
        <View style={styles.logoContainer}>
          <Image resizeMode="contain" source={icon} style={styles.logo} />
        </View>
        <View style={styles.formContainer}>
          <SingleInput placeholder="Single line input" />
          <MultilineInput
            placeholder="Multiline input"
            style={styles.secondInput}
          />
          <MultilineInput placeholder="Large multiline input" />
          <TextInput placeholder="tap me" keyboardType="numeric" />
        </View>
        <View style={styles.submitButtonContainer}>
          <Button onPress={NOOP} title="Submit" />
        </View>
      </ScrollView>
    </SafeAreaView>
  );
};

const styles = StyleSheet.create({
  formContainer: {
    alignItems: "center",
    alignSelf: "stretch",
    justifyContent: "center",
    paddingHorizontal: 10,
  },
  logo: {
    height: 200,
    width: 200,
  },
  logoContainer: {
    alignItems: "center",
    alignSelf: "stretch",
    flex: 1,
    justifyContent: "center",
    paddingVertical: 100,
  },
  secondInput: {
    height: 200,
  },
  submitButtonContainer: {
    alignItems: "center",
    alignSelf: "stretch",
    justifyContent: "center",
    marginTop: 100,
  },
});
```

**Comment by mateusz1913** [comment]
Hi @IvanIhnatsiuk , it's implemented in https://github.com/mateusz1913/react-native-avoid-softinput/commit/96cfc0af23108aaa4526672d11f29dfc0231bd9e and released with version 2.4.0

**Comment by IvanIhnatsiuk** [comment]
Hi @mateusz1913 , Thank you so much for your work! 🚀 🚀
